src/core/heap_size.h

```diff
@@ -29,8 +29,14 @@ namespace dfly {
 
 namespace heap_size_detail {
 
+template <class, class = void> struct has_marked_stackonly : std::false_type {};
+
+template <class T>
+struct has_marked_stackonly<T, std::void_t<typename T::is_stackonly>> : std::true_type {};
+
 template <typename T> constexpr bool StackOnlyType() {
-  return std::is_trivial_v<T> || std::is_same_v<T, std::string_view>;
+  return std::is_trivial_v<T> || std::is_same_v<T, std::string_view> ||
+         has_marked_stackonly<T>::value;
```

Comment:
Wouldn't this be true even for:

```cpp
struct A {
  static constexpr bool is_stackonly = false; 
};
```

Shouldn't we look at the value of `is_stackonly` if exists?

Reply:
yes, that's what I do. but I will try @dranikpg suggestion to replace with trivially destructible

Reply:
I follow similar notion as `is_transparent` for hash functions. it's enough to declare the type, its alias is not important.